internal/server/rule_manager.go

```diff
@@ -75,16 +75,25 @@ func (rr *RuleRegistry) Delete(key string) (*rule.RuleState, bool) {
 	return result, ok
 }
 
-func createRule(name, ruleJson string) (string, error) {
+func createRule(name, ruleJson string) (id string, err error) {
 	var rs *rule.RuleState = nil
-	var err error = nil
+	defer func() {
```

Comment:
This is wrong, should move this to line 105 which the new rule state has been created. Put here may delete the running rule registry